### PR TITLE
feat(swap): Allow to store quote when creating deposit channel through backend [WEB-1557]

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: packages/swap/prisma/schema.prisma [hunk 1]
```diff
@@ -417,23 +418,19 @@ model StateChainError {
...
+  id                         Int                @id @default(autoincrement())
+  swapDepositChannel         SwapDepositChannel @relation(fields: [swapDepositChannelId], references: [id])
+  swapDepositChannelId       BigInt             @unique
+  srcAsset                   InternalAsset
+  destAsset                  InternalAsset
+  depositAmount              Decimal            @db.Decimal(30, 0)
+  intermediateAmount         Decimal?           @db.Decimal(30, 0)
+  egressAmount               Decimal            @db.Decimal(30, 0)
+  estimatedPrice             Decimal
+  executedPrice              Decimal?
```
Comment (niklasnatter) [inline packages/swap/prisma/schema.prisma:431]: will set this in an upcoming pr

File: packages/shared/src/schemas.ts [hunk 2]
```diff
@@ -120,6 +120,13 @@ export const openSwapDepositChannelSchema = z
     srcAddress: z.string().optional(),
     fillOrKillParams: fillOrKillParams.optional(),
     dcaParams: dcaParams.optional(),
+    quote: z
```
Comment (niklasnatter) [inline packages/shared/src/schemas.ts:123]: will pass this from the sdk in an upcoming pr

File: packages/swap/prisma/schema.prisma [hunk 3]
```diff
@@ -417,23 +418,19 @@ model StateChainError {
...
-  quoterFees         Json?
-  poolIntermediate   Decimal?      @db.Decimal(30, 0)
-  poolOutput         Decimal       @db.Decimal(30, 0)
-  poolDuration       Int?
-  poolFees           Json?
-  createdAt          DateTime      @default(now())
-  version            Int
-
-  @@index([srcAsset, destAsset])
+model Quote {
```
Comment (zoheb391) [inline packages/swap/prisma/schema.prisma:421]: wondering if it would be useful to know if it was a boost/dca quote.

Comment (zoheb391) [reply packages/swap/prisma/schema.prisma:421]: without checking the channel

**Review by zoheb391 — APPROVED** [review]

File: packages/swap/prisma/schema.prisma (hunk 3 above)
Comment (niklasnatter) [reply packages/swap/prisma/schema.prisma:421]: yeah true, might be convenient - will add two booleans

File: packages/swap/prisma/schema.prisma (hunk 1 above)
Comment (jerryafr) [reply packages/swap/prisma/schema.prisma:431]: what happens if a channel has been reused? we'd always store the latest executedPrice?

Comment (niklasnatter) [reply packages/swap/prisma/schema.prisma:431]: im think ill store only the first one for now. if its already set, i wont overwrite it

Comment (jerryafr) [reply packages/swap/prisma/schema.prisma:431]: I think we can just store the quote here and a link to the deposit channel already enables us to find the executed swaps. then we can generate the report that we want based on the 3 tables (quote, swapDepositChannel and swapRequest)

**Review by jerryafr — APPROVED** [review]

File: packages/swap/prisma/schema.prisma (hunk 1 above)
Comment (niklasnatter) [reply packages/swap/prisma/schema.prisma:431]: yeah thats true, but its less convenient. dont have a strong opinion though - im also fine with keeping this table to a minimum

File: packages/swap/prisma/schema.prisma (hunk 3 above)
Comment (niklasnatter) [reply packages/swap/prisma/schema.prisma:421]: will do this in a separate pr because i already built on this

**Pull request merged**